EIPS/eip-758.md

```diff
@@ -9,30 +9,32 @@ created: 2017-11-09
 
 ## Specification
 
 ### Subscription
-A caller who wants to be notified of return data for their transactions sends an `eth_subscribe` RPC request with the parameter `"returnData"`:
+A caller who wants to be notified when transactions of theirs complete sends an `eth_subscribe` RPC request with the first parameter `"completedTransaction"`:
 
 ```json
-{"jsonrpc": "2.0", "id": 1, "method": "eth_subscribe", "params": ["returnData"]}
+{"jsonrpc": "2.0", "id": 1, "method": "eth_subscribe", "params": ["completedTransaction", filter]}
 ```
 
+The `filter` parameter is a dictionary containing 3 optional named arguments:  `from`, `to`, and `hasReturnData`.  `from` and `to` can each either be single addresses, or a list of addresses.  They are used to filter out any transactions not sent from an address in the `from` list and sent to an address in the to list.  `hasReturnData` is a boolean--if it is specified and `true`, then notifications will be received only for completed transactions containing returnData.  If none of these optional arguments are present, then the "params" field will be `["completedTransaction", {}]` and no filter restrictions are placed on the transactions.
+
 The Ethereum node responds with a subscription ID:
 
 ```json
 {"jsonrpc": "2.0", "id": 1, "result": "0x00000000000000000000000000000b0b"}
 ```
 
-The caller submits a transaction via `eth_sendTransaction` or `eth_sendRawTransaction` RPC request which has the transaction hash `"0x00000000000000000000000000000000000000000000000000000000deadbeef"`.  When the transaction is sealed (mined), the Ethereum node computes the return value (`"0x000000000000000000000000000000000000000000000000000000000000002a"`) and pushes a notification to the caller:
+The caller submits a transaction via `eth_sendTransaction` or `eth_sendRawTransaction` RPC request which has the transaction hash `"0x00000000000000000000000000000000000000000000000000000000deadbeef"`.  When the transaction is sealed (mined), the Ethereum node pushes a notification to the caller.  If the transaction method call (or contract creation), this will include the return value (`"0x000000000000000000000000000000000000000000000000000000000000002a"`) of the called function:
```

Comment:
I think the last sentence should start with `If the transaction calls a method on a contract...`